llvm/lib/IR/Verifier.cpp

```diff
@@ -812,6 +812,21 @@ void Verifier::visitGlobalValue(const GlobalValue &GV) {
     Check(!GV.hasSection(), "tagged GlobalValue must not be in section.", &GV);
   }
 
+  /// CHERIoT-specific check. If the global value is:
+  /// 1. a global variable
+  /// 2. has the "global capability import" attribute
+  /// then it must be a reference an external object.
+  const auto CheriAttrName =
+      llvm::cheri::GlobalCapabilityImportAttr::getAttrName();
+  if (isa<GlobalVariable>(GV) &&
```

Comment:
The idiomatic way to write this in LLVM is:

```
auto *GV = dyn_cast<GlobalVariable>(GV);
if (GV && GV->hasAttribute(...) {
```

Reply:
[0a4e97e](https://github.com/CHERIoT-Platform/llvm-project/pull/147/commits/0a4e97e7f184f3ecac00072e83b7c82e3f770f09) rewrites it that way.